File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.go
func twoSum(numbers []int, target int) []int {
   left := 0
   right := len(numbers)-1
   for left < right {
	  curr := numbers[left]+numbers[right]
	  if curr == target {
		return []int{left+1,right+1}
	  } else if curr > target {
		right--
	  } else if curr < target {
		left++
	  }
   }
   return []int{-1, -1}
}

File: Data Structures & Algorithms/palindrome-partitioning/submission-1.cpp
class Solution {
public:
	string s;
	vector<string> curr;
	vector<vector<string>> result;
    vector<vector<string>> partition(string s) {
        this->s = s;
		backtrack(0);
		return this->result;
    }

	void backtrack(int i) {
		if (i == s.size()) {
			result.push_back(curr);
			return;
		}

		for (int j = i; j < s.size(); j++) {
			if (!isPalindrome(s, i, j)) {
				continue;
			}

			curr.push_back(s.substr(i, j - i + 1));
			// std::cout << "c: ";
			// for (auto &c : curr) {
			//	std::cout << c << " ";
			// }
			std::cout << std::endl;
			backtrack(j + 1);
			curr.pop_back();
		}
	}

	bool isPalindrome(string s, int left, int right) {
		while (left < right) {
			if (s[left] != s[right]) {
				return false;
			}
			left++;
			right--;
		}
		return true;
	}
};
